### PR TITLE
Refactored bufferManagerImpl to allow hard referenced early buffer

### LINKED ISSUE #2539: Permanent Fix Audit Handler to not lose messages from early buffer
Permanent Fix for #2194

### PR DESCRIPTION
**Pull request by Channyboy**
Refactored bufferManagerImpl to allow an option in the constructor to use a hard referenced early buffer. Modified the unit test as appropriate.

#build 
fixes #2539

### CODE REVIEW

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java [hunk 1]
```diff
@@ -39,33 +39,31 @@
 	protected Queue<Object> earlyMessageQueue;
 
 	private static final int EARLY_MESSAGE_QUEUE_SIZE = 400;
-
+	private final static boolean defaultIsSoftRefEMQ = true;
```
Comment (donbourne) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java]: not sure this is really needed as a variable... suggest getting rid of defaultIsSoftRefEMQ and just update the ctor below to use hardcoded true value.

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java [hunk 2]
```diff
@@ -39,33 +39,31 @@
...
 		this.capacity = capacity;
 		if (!BufferManagerEMQHelper.getEMQRemovedFlag()) {
-			earlyMessageQueue = new SimpleRotatingSoftQueue<Object>(new Object[EARLY_MESSAGE_QUEUE_SIZE]);
-			// Check again just in case
-			if (BufferManagerEMQHelper.getEMQRemovedFlag()) {
-				removeEMQ();
+			if (isSoftRefEMQ){
+				earlyMessageQueue = new SimpleRotatingSoftQueue<Object>(new Object[EARLY_MESSAGE_QUEUE_SIZE]);
+				// Check again just in case
+				if (BufferManagerEMQHelper.getEMQRemovedFlag()) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java]: I think this if block belongs below line 64

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/SimpleRotatingQueue.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
  * Contributors:
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
```
Comment (donbourne) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/SimpleRotatingQueue.java]: update copyright year

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
Failures in latest Personal Build do not relate to the current PR and there already exists defect items for the failures encountered in the latest Personal build.

**Pull request merged**